metadata-db-app/src/cmr/metadata_db/services/generic_documents.clj

```diff
@@ -27,27 +22,21 @@
                   :user-id])
 
 ;; These are fields that should not be inclucded in the :Metadata field because
-;; they are generated by software. Add items to this list corrisponding to 
+;; they are generated by software. Add items to this list corrisponding to
 ;; database column names which should NOT be shared with the world and are not
 ;; in the info-fields list above.
-(def fields-to-drop (into [] (concat info-fields [:id :deleted :transaction-id])))
+(def fields-to-drop (vec (concat info-fields [:id :deleted :transaction-id])))
 
 (defn- raw-generic->response
   "Take the raw database object returned from either SQL or In-Memory and shape
    it into the expected response by pulling out generated fields like concept-id
    and putting them in the :Info field, then putting the metadata inside the
    :Metadata field without any of these extra fields."
   [raw-doc]
-  {:Info (assoc-these-from info-fields raw-doc)
+  {:Info (select-keys raw-doc info-fields)
```

Comment:
I agree with your choice to use select-keys here for readability, but just want to make sure -- your previous function included cutil/remove-nil-keys, do you not need that anymore? I don't think select-keys covers that but I may be wrong.